stacker/hooks/keypair.py

```diff
@@ -0,0 +1,49 @@
+import logging
+import os
+
+from boto.ec2 import connect_to_region
+
+logger = logging.getLogger(__name__)
+
+
+def ensure_keypair_exists(region, namespace, mappings, parameters, **kwargs):
+    connection = connect_to_region(region)
+    keypair_name = kwargs.get('keypair', parameters.get('SshKeyName'))
+    keypair = connection.get_key_pair(keypair_name)
+    message = 'keypair: %s (%s) %s'
+    if keypair:
+        logger.info(message, keypair.name, keypair.fingerprint, 'exists')
+        return True
+
+    logger.info('keypair: "%s" not found', keypair_name)
+    create_or_upload = raw_input(
+        'import or create keypair "%s"? (import/create/cancel) ' % (
+            keypair_name,
+        ),
+    )
+    if create_or_upload == 'import':
+        path = raw_input('path to keypair file: ')
+        full_path = os.path.abspath(os.path.expanduser(path))
```

Comment:
Should you use your full path thing that you added to util in the other PR?  Kind of annoying, I know.